src/ballet/chacha20/test_fd_chacha20_rng.c

```diff
@@ -0,0 +1,60 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+#include "fd_chacha20.h"
+
+#define TEST_KEY_SIZE FD_CHACHA20_KEY_SIZE
+#define TEST_NONCE_SIZE FD_CHACHA20_NONCE_SIZE
+
+/* Test fd_chacha20_rng_init function */
+void test_fd_chacha20_rng_init()
+{
+  unsigned char key[TEST_KEY_SIZE];
+  unsigned char nonce[TEST_NONCE_SIZE];
+
+  /* Generate key and nonce using /dev/urandom */
+  FILE *fp = fopen("/dev/urandom", "rb");
+  fread(key, sizeof(key), 1, fp);
+  fread(nonce, sizeof(nonce), 1, fp);
+  fclose(fp);
+
+  /* Initialize the random number generator */
+  assert(fd_chacha20_rng_init(key, nonce) == 0);
+}
+
+/* Test fd_chacha20_rng_get_uint32 function */
+void test_fd_chacha20_rng_get_uint32()
+{
+  unsigned char key[TEST_KEY_SIZE];
+  unsigned char nonce[TEST_NONCE_SIZE];
+  fd_chacha20_rng_t num1, num2;
+
+  /* Generate key and nonce using /dev/urandom */
+  FILE *fp = fopen("/dev/urandom", "rb");
+  fread(key, sizeof(key), 1, fp);
+  fread(nonce, sizeof(nonce), 1, fp);
+  fclose(fp);
+
+  /* Initialize the random number generator */
+  assert(fd_chacha20_rng_init(key, nonce) == 0);
```

Comment:
We normally prefer `FD_TEST` over `assert` for a unit test

Reply:
updated ✅